include/pybind11/attr.h

```diff
@@ -14,27 +14,22 @@
 
 NAMESPACE_BEGIN(pybind11)
 
-template <typename T> struct arg_t;
-
 /// Annotation for keyword arguments
 struct arg {
     constexpr explicit arg(const char *name) : name(name) { }
 
     template <typename T>
     constexpr arg_t<T> operator=(const T &value) const { return {name, value}; }
-    template <typename T, size_t N>
-    constexpr arg_t<const T *> operator=(T const (&value)[N]) const {
-        return operator=((const T *) value);
-    }
 
     const char *name;
 };
 
 /// Annotation for keyword arguments with default values
 template <typename T> struct arg_t : public arg {
     constexpr arg_t(const char *name, const T &value, const char *descr = nullptr)
-        : arg(name), value(value), descr(descr) { }
-    T value;
+        : arg(name), value(&value), descr(descr) { }
```

Comment:
Switching to `py::object` would be the best case. There would be no restrictions on `py::arg_t` then. As a bonus, the template code would also be simplified since `py::arg_t` would no longer be a template class (it would only have a template constructor). `py::arg_t` should create it's `py::object` using the `automatic` policy which looks like the only reasonable choice. 

There is just one small inconsistency that this would introduce: Keyword arguments would then be passed to function calls with the `automatic` policy, while positional arguments would use `automatic_reference` (which is the function call default). I don't think this a big issue, but maybe you have some thoughts.